lane.py

```diff
@@ -347,10 +349,9 @@ def apply(self, _db, qu, distinct):
         from model import MaterializedWorkWithGenre as work_model
         quality = self.quality_tier_field()
         qu = qu.order_by(
-            quality.desc(), work_model.random.desc()
+            quality.desc(), work_model.random.desc(), work_model.works_id
         )
-        if distinct:
-            qu = qu.distinct(work_model.works_id)
+        qu = qu.distinct(quality, work_model.random, work_model.works_id)
```

Comment:
Why are you setting the query distinct on multiple values here?

Reply:
If you order a query by X, you can set it distinct on (X, Y) but you can't set it distinct solely on Y. The query here is ordered by quality, then random, so I have to put those first in the distinct() calls.

I think the code I'm replacing either doesn't work, or only works in a limited situation. Here's the corresponding code currently in master, which sets DISTINCT on the same fields used in ORDER BY.

```
        # Set the ORDER BY clause.
        order_by, order_distinct = self.order_by(work_model)
        qu = qu.order_by(*order_by)
        if distinct:
            qu = qu.distinct(*order_distinct)
```

Reply:
Gotcha, thanks for explaining.